### PR TITLE
Export testlab based on Node version

### PR DESCRIPTION
**Pull request by superkhau**

### CODE REVIEW

File: packages/testlab/package.json [hunk 1]
```diff
@@ -4,10 +4,12 @@
...
   "main": "index",
   "scripts": {
-    "build": "tsc && npm run build:es2015",
-    "build:es2015": "tsc --target es2015 --outDir lib-es2015",
+    "build": "npm run build:es2015 && npm run build:es2017",
+    "build:es2015": "tsc --target es2015 --outDir lib6",
+    "build:es2017": "tsc --target es2017 --outDir lib",
     "clean": "rm *.tgz; rm -rf ./package",
     "prepublish": "npm run build",
+    "pretest": "npm run build",
```
Comment (superkhau) [inline packages/testlab/package.json]: @bajtos This should address https://github.com/strongloop/loopback-next/pull/225#issuecomment-297686889 for now?

File: packages/testlab/package.json [hunk 2]
```diff
@@ -4,10 +4,12 @@
   "description": "A collection of test utilities we use to write LoopBack tests.",
   "main": "index",
   "scripts": {
-    "build": "tsc && npm run build:es2015",
-    "build:es2015": "tsc --target es2015 --outDir lib-es2015",
+    "build": "npm run build:es2015 && npm run build:es2017",
+    "build:es2015": "tsc --target es2015 --outDir lib6",
+    "build:es2017": "tsc --target es2017 --outDir lib",
```
Comment (superkhau) [inline packages/testlab/package.json]: @bajtos This should address https://github.com/strongloop/loopback-next/pull/225#discussion_r113669953.

Annoyingly, the `--outDir` must be `'es3', 'es5', 'es6', 'es2015', 'es2016', 'es2017', 'esnext'` (so we can't have `es7` for example).

Comment (superkhau) [inline packages/testlab/package.json]: Might be smart to go with lib2015 for consistency, I'm OK either way (I just mapped it to lib6 for now).

File: packages/testlab/package.json [hunk 3]
```diff
@@ -4,10 +4,12 @@
   "description": "A collection of test utilities we use to write LoopBack tests.",
   "main": "index",
   "scripts": {
-    "build": "tsc && npm run build:es2015",
-    "build:es2015": "tsc --target es2015 --outDir lib-es2015",
+    "build": "npm run build:es6 && npm run build:es8",
+    "build:es6": "tsc --target es2015 --outDir lib6",
+    "build:es8": "tsc --target es2017 --outDir lib",
```
Comment (superkhau) [inline packages/testlab/package.json]: I updated the code to be more consistent/simple.

Comment (bajtos) [inline packages/testlab/package.json]: Nitpick - feel free to ignore:

I don't think `es8` is a thing - we have es6 = es2015 and then they switched to year-based versioning -
 es2016, es2017, etc.

Maybe `build:lib` and `build:lib6` would make the intent more clear?

File: packages/testlab/package.json (hunk 1 above)
Comment (bajtos) [inline packages/testlab/package.json]: Yes, thanks 👍

File: packages/testlab/package.json (hunk 2 above)
Comment (bajtos) [inline packages/testlab/package.json]: You mean `--target`, right? See my https://github.com/strongloop/loopback-next/pull/236#discussion_r113874485, `es7` does not exist.

File: packages/testlab/package.json (hunk 3 above)
Comment (superkhau) [inline packages/testlab/package.json]: I was going with wikipedia versions -- https://en.wikipedia.org/wiki/ECMAScript (ES8 really isn't a thing and there are too many aliases. I'm going to go with build:lib + lib6 as you suggested for clarity.

**Comment by bajtos** [comment]
>  I'm going to go with build:lib + lib6 as you suggested for clarity.

Great, thank you 👍

**Pull request merged**